service/history/api/create_workflow_util.go

```diff
@@ -67,7 +68,7 @@ func NewWorkflowWithSignal(
 	runID string,
 	startRequest *historyservice.StartWorkflowExecutionRequest,
 	signalWithStartRequest *workflowservice.SignalWithStartWorkflowExecutionRequest,
-) (WorkflowLease, error) {
+) (workflow.MutableState, error) {
```

Comment:
Returning only the MS instead of the lease allows the caller to construct the lease.